src/main/java/org/cascadebot/cascadebot/utils/interactions/InteractionCache.java

```diff
@@ -3,37 +3,38 @@
  * Licensed under the MIT license.
  */
 
-package org.cascadebot.cascadebot.utils.buttons;
+package org.cascadebot.cascadebot.utils.interactions;
 
 import org.cascadebot.cascadebot.utils.DiscordUtils;
+import org.cascadebot.cascadebot.utils.interactions.ComponentContainer;
 
 import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.Map;
 
-public class ButtonsCache extends HashMap<Long, LinkedHashMap<Long, ButtonGroup>> { //Long 1 is channel id. long 2 is message id. This uses a linked hashmap so we can remove the oldest button when we get pasted 5 buttons per channel
+public class InteractionCache extends HashMap<Long, LinkedHashMap<Long, ComponentContainer>> { //Long 1 is channel id. long 2 is message id. This uses a linked hashmap so we can remove the oldest button when we get pasted 5 buttons per channel
```

Comment:
Instead of inheriting HashMap, could the HashMap be stored in a field instead?

Just because we're not overriding all functions, there could be unexpected side effects if someone calls the wrong method accidentally. 

Composition over inheritance

Reply:
We're overriding literally 0 functions of the map, this extends from maps so things accessing it can easily navigate it. I don't think the issue of calling the wrong method is a big deal as we'll catch it in the pr, and really the only method wrong you could call is put(). I can still switch it to a separate variable, but I think it's better extending honestly.